partition_snapshot_reader.hh

```diff
@@ -199,17 +199,17 @@ class partition_snapshot_flat_reader : public flat_mutation_reader::impl, public
             return !_range_tombstones.empty();
         }
     public:
-        explicit lsa_partition_reader(const schema& s, reader_permit permit, partition_snapshot_ptr snp,
+        explicit lsa_partition_reader(const schema_ptr& s, reader_permit permit, partition_snapshot_ptr snp,
                                       logalloc::region& region, logalloc::allocating_section& read_section,
```

Comment:
I guess if schema::make_reversed() returned const schema& this wouldn't be needed.

Reply:
If the issue here is with the parameter type (and not with the fact, that we store schema_ptr in lsa_partition_reader instead of const schema&), then even with the current behaviour of make_reversed, I could leave the `const schema& s` as parameter and use
```
            , _snapshot_schema(Reversing ? s.make_reversed() : s.shared_from_this())
```